client/my-sites/github-deployments/controller.tsx

```diff
@@ -0,0 +1,34 @@
+import { isEnabled } from '@automattic/calypso-config';
+import { PageViewTracker } from 'calypso/lib/analytics/page-view-tracker';
+import isAtomicSite from 'calypso/state/selectors/is-site-automated-transfer';
+import { isJetpackSite } from 'calypso/state/sites/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import { GitHubDeployments } from './main';
+import type { Callback } from '@automattic/calypso-router';
+
+export const githubDeployments: Callback = ( context, next ) => {
+	context.primary = (
+		<>
+			<PageViewTracker path="/github-deployments/:site" title="Github Deployments" delay={ 500 } />
+			<GitHubDeployments />
+		</>
+	);
+	next();
+};
+
+export const redirectHomeIfIneligible: Callback = ( context, next ) => {
+	const state = context.store.getState();
+	const siteId = getSelectedSiteId( state );
+
+	if ( ! isEnabled( 'github-integration-i1' ) || ! isAtomicSite( state, siteId ) ) {
```

Comment:
Let's move forward with that in this PR, but create a follow up to transfer the site to Atomic if eligible, a la `/hosting-config/%s`.

Reply:
Yeah we probably gonna show an upsell or something